internal/provider/search_attribute_resource.go

```diff
@@ -0,0 +1,292 @@
+		Namespace: state.Namespace.ValueString(),
+	})
+
+	if err != nil {
+		resp.Diagnostics.AddError("Client Error", fmt.Sprintf("Unable to read search attribute info, got error: %s", err))
+		return
+	}
+
+	tflog.Trace(ctx, "Read a Temporal search attribute resource")
+
+	// Attempt to find search attribute
+	attr, ok := attributes.CustomAttributes[state.Name.ValueString()]
+	if !ok {
+		// Delete resource if not found in underlying system
+		tflog.Info(ctx, "Resource not found, removing from state")
+		resp.State.RemoveResource(ctx)
+		return
+	}
+
+	data := &SearchAttributeResourceModel{
+		Name:      state.Name,
+		Namespace: state.Namespace,
+		Type:      types.StringValue(attr.String()),
+	}
+
+	// Set refreshed state
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+// Update is a no-op method that handles update calls without making any changes.
+func (r *SearchAttributeResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+
+	tflog.Warn(ctx, "Update operation called, but updates are not supported for this resource.")
+
+}
+
+// Delete removes a Temporal search attribute from both Temporal and the Terraform state.
+func (r *SearchAttributeResource) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	var data SearchAttributeResourceModel
+
+	client := operatorservice.NewOperatorServiceClient(r.client)
+
+	// Read Terraform prior state data into the model
+	resp.Diagnostics.Append(req.State.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Delete request
+	request := &operatorservice.RemoveSearchAttributesRequest{
+		Namespace:        data.Namespace.ValueString(),
+		SearchAttributes: []string{data.Name.ValueString()},
+	}
+
+	_, err := client.RemoveSearchAttributes(ctx, request)
+
+	if err != nil {
+		if _, ok := err.(*serviceerror.NotFound); ok {
+			resp.Diagnostics.AddError("Request error", "Search attribute not found: "+err.Error())
+			return
+		}
+		resp.Diagnostics.AddError("Request error", "Unable to delete search attribute "+err.Error())
+		return
+	}
+
+	tflog.Info(ctx, fmt.Sprintf("Successfully deleted search attribute: %s", data.Name.ValueString()))
+	tflog.Trace(ctx, "Deleted a resource")
+}
+
+// ImportState allows existing Temporal search attributes to be imported into the Terraform state.
+// Importing system attributes is not supported
+func (r *SearchAttributeResource) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+
+	// Expected request ID format is a string 'namespace:search_attribute_name'
+	// Ex: 'default:CustomBool'
+
+	parts := strings.Split(req.ID, ":")
+	if len(parts) < 2 {
+		resp.Diagnostics.AddError("Invalid ID format", "Expected 'namespace:name'.")
+
+		return
+	}
+
+	namespace := parts[0]
+	name := parts[1]
```

Comment:
Add some documentation here with details of what is in parts[0] and what is in parts[1]